.gitlab/docker/Dockerfile.spack_base

```diff
@@ -25,9 +25,6 @@ RUN apt-get update && apt-get -yqq install --no-install-recommends \
     openssl \
     pipx \
     pkgconf \
-    python3 \
-    python3-dev \
-    python3-pip \
```

Comment:
Note that python is installed even if you remove these: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/479009878135925/5304355110917878/-/jobs/6173414941#L5232. Presumably because of pipx?

I would probably do the following:
- Keep them deleted here if we don't need it _explicitly_ otherwise. Since pipx uses python it can request installation of python, but if we remove pipx so should python.
- Instead do `spack external find --exclude python` to let spack use its own python.
- Remove the workaround for meson (at least if I understand correctly that this workaround is only required when using an external python).